src/monattack.cpp

```diff
@@ -560,6 +560,56 @@ void mattack::boomer(monster *z, int index)
     }
 }
 
+void mattack::boomer_glow(monster *z, int index)
+{
+    if( !z->can_act() ) {
+        return;
+    }
+
+    int t;
+    Creature *target = z->attack_target();
+    if( target == nullptr ||
+        rl_dist( z->pos(), target->pos() ) > 3 ||
+        !z->sees( *target, t ) ) {
+        return;
+    }
+
+    player *foe = dynamic_cast< player* >( target );
+    std::vector<tripoint> line = line_to( z->pos(), target->pos(), t, 0 );
+    z->reset_special(index); // Reset timer
+    z->moves -= 250;   // It takes a while
+    bool u_see = g->u.sees( *z );
+    if( u_see ) {
+        add_msg(m_warning, _("The %s spews bile!"), z->name().c_str());
+    }
+    for (auto &i : line) {
+        g->m.add_field(i.x, i.y, fd_bile, 1);
+        if (g->m.move_cost(i.x, i.y) == 0) {
```

Comment:
Use `i` instead of `i.x, i.y` there and in all similar `map` functions. The x/y overloads will be removed soon and all those functions accept tripoints.